scripts/lldb.js

```diff
@@ -13,43 +14,103 @@ function versionToBranch(version) {
+ * @returns {string} Path to the lldb API headers
  */
-function getHeadersPath(lldbInstallDir) {
-  return path.join(lldbInstallDir, 'include', 'lldb');
+function getApiHeadersPath(includeDir) {
+  return path.join(includeDir, 'lldb', 'API');
+}
+
+/**
+ * @param {string} includeDir Path to the equivalent of llvm-config --libddir
+ * @returns {string} Path to the lldb shared library
+ */
+function getLibPath(libDir) {
+  const lib = os.type() === 'Darwin' ? 'liblldb.dylib' : 'liblldb.so';
+  return path.join(libDir, lib);
 }
 
 /**
  * Check out source code of the lldb for headers
  * TODO: The llvm project is probably moving to github soon at that point we
  * should stop using the mirror.
  * @param {string} lldbVersion Version of lldb, either like 3.9 or 39
- * @param {string} buildDir
- * @returns {string} The directory where the source code is checked out
+ * @param {string} buildDir Path to the llnode module directory
+ * @returns {string} The include directory in the downloaded lldb source code
  */
 function cloneHeaders(lldbVersion, buildDir) {
   const lldbHeadersBranch = versionToBranch(lldbVersion);
-  const lldbInstallDir = `lldb-${lldbVersion}`;
-
-  if (!fs.existsSync(path.join(buildDir, lldbInstallDir))) {
-    console.log(`Cloning lldb ${lldbHeadersBranch} into ${lldbInstallDir}`);
-    child_process.execFileSync('git',
-        ['clone', '--depth=1', '-b', lldbHeadersBranch,
-          'https://github.com/llvm-mirror/lldb.git', lldbInstallDir],
-        {
-          cwd: buildDir,
-          stdio: 'inherit'  // show progress
-        });
+  const lldbInstallDir = path.resolve(buildDir, `lldb-${lldbVersion}`);
+
+  if (!fs.existsSync(lldbInstallDir)) {
+    console.log(`\nCloning lldb ${lldbHeadersBranch} into ${lldbInstallDir}`);
+    child_process.execFileSync(
+        'git', ['clone',
+          '--depth', '1',
+          '--branch', lldbHeadersBranch,
+          'https://github.com/llvm-mirror/lldb.git',
+          lldbInstallDir
+        ],
+        { stdio: 'inherit' });  // show progress
   } else {
-    console.log(`Skip cloning lldb headers because ${lldbInstallDir} exists`);
+    console.log(`\nSkip cloning lldb headers because ${lldbInstallDir} exists`);
   }
-  return lldbInstallDir;
+  return path.join(lldbInstallDir, 'include');
+}
+
+/**
+ * Try to find the first valid executable out of an array executable names.
+ * Returns undefined if none of the provided executables is valid, otherwise
+ * returns the path to the first found valid executable.
+ * @param {string[]} exeNames
+ * @returns {string|undefined}
+ */
+function tryExecutables(exeNames) {
+  let exePath;
+  for (let name of exeNames) {
+    try {
+      exePath = child_process.execFileSync(
+          'which', [name], { stdio: 'pipe' }  // to suppress stderr
+      ).toString().trim();
+      // If the result starts with '/' `which` found a path.
+      if (exePath.startsWith('/')) {
+        return exePath;
+      }
```

Comment:
Everything except the `.execFileSync()` call should really be moved outside the try/catch block and the catch block arguably should check that `err` is what it expects it to be.

(That advice applies to other try/catch blocks as well - keep the try block as small as possible.)